AppCache removal: remove manifest attribute

annevk opened the pull request:
#### f5bdfbfc1c90bdfb897d9d1d03827854ec3128bf
<pre>
AppCache removal: remove manifest attribute
<a href="https://bugs.webkit.org/show_bug.cgi?id=275251">https://bugs.webkit.org/show_bug.cgi?id=275251</a>

Reviewed by Youenn Fablet.

As AppCache has been fully disabled it&apos;s time to remove the code.

* Source/WebCore/dom/Document.cpp:
(WebCore::Document::hasManifest const): Deleted.
* Source/WebCore/dom/Document.h:
* Source/WebCore/html/HTMLAttributeNames.in:
* Source/WebCore/html/HTMLHtmlElement.cpp:
(WebCore::HTMLHtmlElement::isURLAttribute const): Deleted.
(WebCore::HTMLHtmlElement::insertedByParser): Deleted.
* Source/WebCore/html/HTMLHtmlElement.h:
* Source/WebCore/html/ImageDocument.cpp:
(WebCore::ImageDocument::createDocumentStructure):
* Source/WebCore/html/MediaDocument.cpp:
(WebCore::MediaDocumentParser::createDocumentStructure):
* Source/WebCore/html/ModelDocument.cpp:
(WebCore::ModelDocumentParser::createDocumentStructure):
* Source/WebCore/html/PDFDocument.cpp:
(WebCore::PDFDocument::createDocumentStructure):
* Source/WebCore/html/PluginDocument.cpp:
(WebCore::PluginDocumentParser::createDocumentStructure):
* Source/WebCore/html/parser/HTMLConstructionSite.cpp:
(WebCore::HTMLConstructionSite::insertHTMLHtmlStartTagBeforeHTML):
* Source/WebCore/loader/DocumentLoader.cpp:
(WebCore::DocumentLoader::finishedLoading):
* Source/WebCore/xml/parser/XMLDocumentParserLibxml2.cpp:
(WebCore::XMLDocumentParser::startElementNs):
* Source/WebKitLegacy/mac/DOM/DOMHTMLHtmlElement.mm:
(-[DOMHTMLHtmlElement manifest]):
(-[DOMHTMLHtmlElement setManifest:]):

Canonical link: <a href="https://commits.webkit.org/279811@main">https://commits.webkit.org/279811@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/80e041d89b2b49ab10f2dd6248dc8c404dbc493d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/54627 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/34049 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/7203 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/57908 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/5361 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/56929 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/41587 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/5366 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/44252 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/3617 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/56722 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/32192 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/47316 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/25378 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/28989 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/4650 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/3501 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/50737 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/4864 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/59498 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/29864 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/5006 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/51671 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/31010 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/47402 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/51055 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/31994 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8078 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/30798 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

youennf reviewed (approved):
If we can remove HTMLNames manifest by making the WK1 methods no-ops, let's do it.

annevk committed:
AppCache removal: remove manifest attribute
https://bugs.webkit.org/show_bug.cgi?id=275251

Reviewed by Youenn Fablet.

As AppCache has been fully disabled it's time to remove the code.

* Source/WebCore/dom/Document.cpp:
(WebCore::Document::hasManifest const): Deleted.
* Source/WebCore/dom/Document.h:
* Source/WebCore/html/HTMLAttributeNames.in:
* Source/WebCore/html/HTMLHtmlElement.cpp:
(WebCore::HTMLHtmlElement::isURLAttribute const): Deleted.
(WebCore::HTMLHtmlElement::insertedByParser): Deleted.
* Source/WebCore/html/HTMLHtmlElement.h:
* Source/WebCore/html/ImageDocument.cpp:
(WebCore::ImageDocument::createDocumentStructure):
* Source/WebCore/html/MediaDocument.cpp:
(WebCore::MediaDocumentParser::createDocumentStructure):
* Source/WebCore/html/ModelDocument.cpp:
(WebCore::ModelDocumentParser::createDocumentStructure):
* Source/WebCore/html/PDFDocument.cpp:
(WebCore::PDFDocument::createDocumentStructure):
* Source/WebCore/html/PluginDocument.cpp:
(WebCore::PluginDocumentParser::createDocumentStructure):
* Source/WebCore/html/parser/HTMLConstructionSite.cpp:
(WebCore::HTMLConstructionSite::insertHTMLHtmlStartTagBeforeHTML):
* Source/WebCore/loader/DocumentLoader.cpp:
(WebCore::DocumentLoader::finishedLoading):
* Source/WebCore/xml/parser/XMLDocumentParserLibxml2.cpp:
(WebCore::XMLDocumentParser::startElementNs):
* Source/WebKitLegacy/mac/DOM/DOMHTMLHtmlElement.mm:
(-[DOMHTMLHtmlElement manifest]):
(-[DOMHTMLHtmlElement setManifest:]):

Canonical link: https://commits.webkit.org/279811@main
